secrets/azurekeyvault/akv_test.go

```diff
@@ -127,10 +125,12 @@ func initEnv() {
 	// For Client Certificate and Azure Managed Service Identity, see doc below for help
 	// https://github.com/Azure/azure-sdk-for-go
 
-	if os.Getenv("AZURE_TENANT_ID") == "" ||
-		os.Getenv("AZURE_CLIENT_ID") == "" ||
-		os.Getenv("AZURE_CLIENT_SECRET") == "" {
-		log.Fatal("Missing environment for recording tests, set AZURE_TENANT_ID, AZURE_CLIENT_ID and AZURE_CLIENT_SECRET")
+	if *setup.Record {
```

Comment:
Is this change because `AZURE_ENVIRONMENT` is required for the code to work even during replay? Why? It's not used or document anywhere in our code.

Reply:
We don't need to set it for either replay/record, it is set at L119 above using a default value "AZUREPUBLICCLOUD". We do mention in the prerequisites comment at the top of the file for how to setup for record.

Do you think it is necessary to make this also read from environment variable instead of hard-coded? For reference the available values are:
```go
var environments = map[string]Environment{
	"AZURECHINACLOUD":        ChinaCloud,
	"AZUREGERMANCLOUD":       GermanCloud,
	"AZUREPUBLICCLOUD":       PublicCloud,
	"AZUREUSGOVERNMENTCLOUD": USGovernmentCloud,
}
```

Reply:
No, I think `AZUREPUBLICCLOUD` is fine.

I still don't understand why you changed the code from only calling `initEnv` in `--record` mode to always calling it.

Reply:
I think I was having bad local env value and azure was using that, it is not needed, changed it back.